### PR TITLE
Add support for space-supporter role for set-space-role and unset-space-role

### PR DESCRIPTION
**Pull request by JenGoldstrich**
## Does this PR modify CLI v6 or v7?

v8!

## Description of the Change

Adds support for space-supporter role on set-space-role and unset-space-role, it also makes role entry for these commands both case insensitive.

Help text needs to be modified, will ask in comments of this PR for what the folks working on RBAC want it to say

This will throw an error if you run it on any foundation not set up for space-supporter, which is most, so the command needs to be listed as beta and subject to change

## Applicable Issues

https://github.com/cloudfoundry/cli/issues/2175

### CODE REVIEW

**Comment by JenGoldstrich** [comment]
@monamohebbi What should the help text for set-space-role and unset-space-role?

**Comment by JenGoldstrich** [comment]
@monamohebbi I have requested your review on this PR, once that's done it should be good to merge!

File: integration/v7/isolated/unset_space_role_command_test.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ var _ = Describe("unset-space-role command", func() {
 				Eventually(session).Should(Say("SpaceManager - Invite and manage users, and enable features for a given space"))
 				Eventually(session).Should(Say("SpaceDeveloper - Create and manage apps and services, and see logs and reports"))
 				Eventually(session).Should(Say("SpaceAuditor - View logs, reports, and settings on this space"))
+				Eventually(session).Should(Say("SpaceSupporter -"))
```
Comment (monamohebbi) [inline integration/v7/isolated/unset_space_role_command_test.go:45]: Could this test be updated to include the help text you added in the next commit?

File: command/v7/set_space_role_command.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ type SetSpaceRoleCommand struct {
...
-	usage           interface{}        `usage:"CF_NAME set-space-role USERNAME ORG SPACE ROLE\n   CF_NAME set-space-role USERNAME ORG SPACE ROLE [--client]\n   CF_NAME set-space-role USERNAME ORG SPACE ROLE [--origin ORIGIN]\n\nROLES:\n   SpaceManager - Invite and manage users, and enable features for a given space\n   SpaceDeveloper - Create and manage apps and services, and see logs and reports\n   SpaceAuditor - View logs, reports, and settings on this space\n   SpaceSupporter - (Beta role, behavior will change) TEMPLATE MESSAGE"`
+	usage           interface{}        `usage:"CF_NAME set-space-role USERNAME ORG SPACE ROLE\n   CF_NAME set-space-role USERNAME ORG SPACE ROLE [--client]\n   CF_NAME set-space-role USERNAME ORG SPACE ROLE [--origin ORIGIN]\n\nROLES:\n   SpaceManager - Invite and manage users, and enable features for a given space\n   SpaceDeveloper - Create and manage apps and services, and see logs and reports\n   SpaceAuditor - View logs, reports, and settings on this space\n   SpaceSupporter [Beta role, subject to change] - Manage apps and services, and see logs and reports"`
```
Comment (monamohebbi) [inline command/v7/set_space_role_command.go:19]: I like this message, but I'm starting to overthink a bit. Thoughts on this wording for the first half:

Manage app lifecycle and service bindings

Open to changing this though if this isn't what app lifecycle means to you.

**Review by monamohebbi — CHANGES_REQUESTED** [review]
Looks good! I'm happy with the help text you have and don't think we should get to deep into bikeshedding, but let me know your thoughts on my suggestions. But could you update the test for whatever text you end up using?

File: integration/v7/isolated/unset_space_role_command_test.go (hunk 1 above)
Comment (jdgonzaleza) [reply integration/v7/isolated/unset_space_role_command_test.go]: Updated this in the newest commit

File: command/v7/set_space_role_command.go (hunk 2 above)
Comment (jdgonzaleza) [reply command/v7/set_space_role_command.go]: Updated this in the newest commit

**Review by monamohebbi — APPROVED** [review]
LGTM!

File: command/v7/set_space_role_command.go [hunk 3]
```diff
@@ -89,13 +91,15 @@ func (cmd SetSpaceRoleCommand) validateFlags() error {
 }
 
 func convertSpaceRoleType(givenRole flag.SpaceRole) (constant.RoleType, error) {
-	switch givenRole.Role {
-	case "SpaceAuditor":
+	switch strings.ToLower(givenRole.Role) {
```
Comment (a-b) [inline command/v7/set_space_role_command.go:94]: Love that!

**Review by a-b — APPROVED** [review]
LGTM

**Pull request merged**